src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -16,66 +18,151 @@
+      @RequestParam("semester") String semester,
+      @RequestParam("studentId") int studentId) {
+
+    User student = userRepository.findStudentByStudentId(studentId);
+    if (student == null) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+          "Student not found for id: " + studentId);
+    }
+    List<Enrollment> enrollments = enrollmentRepository.findByYearAndSemesterOrderByCourseId(year, semester, studentId);
+    return getEnrollmentDTOS(enrollments);
+  }
+
+
+  // student adds enrollment into a section
+  // user must be student
+  // return EnrollmentDTO with enrollmentId generated by database
+  @PostMapping("/enrollments/sections/{sectionNo}")
+  public EnrollmentDTO addCourse(
+      @PathVariable int sectionNo,
+      @RequestParam("studentId") int studentId ) {
+
+    Section section = sectionRepository.findSectionBySectionNo(sectionNo);
+    if (section.getCourse().getTitle().isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "section not found: " + sectionNo);
+    }
 
-   // student gets transcript showing list of all enrollments
-   // studentId will be temporary until Login security is implemented
-   //example URL  /transcript?studentId=19803
-   @GetMapping("/transcripts")
-   public List<EnrollmentDTO> getTranscript(@RequestParam("studentId") int studentId) {
-
-       // TODO
-
-       // list course_id, sec_id, title, credit, grade in chronological order
-       // user must be a student
-	   // hint: use enrollment repository method findEnrollmentByStudentIdOrderByTermId
-       // remove the following line when done
-       return null;
-   }
-
-    // student gets a list of their enrollments for the given year, semester
-    // user must be student
-    // studentId will be temporary until Login security is implemented
-   @GetMapping("/enrollments")
-   public List<EnrollmentDTO> getSchedule(
-           @RequestParam("year") int year,
-           @RequestParam("semester") String semester,
-           @RequestParam("studentId") int studentId) {
-
-
-     // TODO
-	 //  hint: use enrollment repository method findByYearAndSemesterOrderByCourseId
-     //  remove the following line when done
-       return null;
-   }
+    long time_ms = System.currentTimeMillis();
+    java.sql.Date today = new java.sql.Date(time_ms);
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-mm-dd");
+    Date addDate = section.getTerm().getAddDate();
+    Date addDeadline = section.getTerm().getAddDeadline();
+    if(today.before(addDate) || today.after(addDeadline)) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Date Invalid");
+    }
 
+    List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsByStudentIdOrderByTermId(studentId);
+    List<EnrollmentDTO> dtoEnrollments = new ArrayList<>();
+    if(!enrollments.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Student is already enrolled");
```

Comment:
This will be true if the student is enrolled in any course, not just the one being added.

Reply:
Fixed via commit [8ee9dd7](https://github.com/jdolarcsumb/CST438-Assignment2-02/commit/8ee9dd7028c2f4e36b7f0912a2fbf4ed8110a42b)